src/java.base/share/classes/java/util/FormatProcessor.java

```diff
@@ -0,0 +1,284 @@
+ * FormatProcessor fmt = FormatProcessor.create(Locale.ROOT);
+ * int x = 10;
+ * int y = 20;
+ * String result1 = fmt."\{x} + \{y} = \{x + y}";
+ * String result2 = fmt."%s\{x} + %s\{y} = %s\{x + y}";
+ * }
+ * In the above example, the value of {@code result1} and {@code result2} will
+ * both be {@code "10 + 20 = 30"}.
+ * <p>
+ * {@link FormatProcessor}  format specification uses and exceptions are the same as
+ * those of {@link Formatter}.
+ * <p>
+ * However, there are two significant differences related to the position of arguments.
+ * An explict {@code n$} and relative {@code <} index will cause an exception due to
+ * a missing argument list.
+ * Whitespace appearing between the specification and the embedded expression will
+ * also cause an exception.
+ * <p>
+ * {@link FormatProcessor} allows the use of different locales. For example:
+ * {@snippet :
+ * Locale locale = Locale.forLanguageTag("th-TH-u-nu-thai");
+ * FormatProcessor thaiFMT = FormatProcessor.create(locale);
+ * int x = 10;
+ * int y = 20;
+ * String result = thaiFMT."%d\{x} + %d\{y} = %d\{x + y}";
+ * }
+ * In the above example, the value of {@code result} will be
+ * {@code "\u0E51\u0E50 + \u0E52\u0E50 = \u0E53\u0E50"}.
+ * <p>
+ * For day to day use, the predefined {@link FormatProcessor#FMT} {@link FormatProcessor}
+ * is available. {@link FormatProcessor#FMT} is defined using the {@link Locale#ROOT}.
+ * Example: {@snippet :
+ * int x = 10;
+ * int y = 20;
+ * String result = FMT."0x%04x\{x} + 0x%04x\{y} = 0x%04x\{x + y}"; // @highlight substring="FMT"
+ * }
+ * In the above example, the value of {@code result} will be {@code "0x000a + 0x0014 = 0x001E"}.
+ *
+ * @since 21
+ *
+ * @see java.lang.template.StringProcessor
+ */
+@PreviewFeature(feature=PreviewFeature.Feature.STRING_TEMPLATES)
+public final class FormatProcessor implements StringProcessor, ProcessorLinkage {
+    /**
+     * {@link Locale} used to format
+     */
+    private final Locale locale;
+
+    /**
+     * Constructor.
+     *
+     * @param locale  {@link Locale} used to format
+     */
+    private FormatProcessor(Locale locale) {
+        this.locale = locale;
+    }
+
+    /**
+     * Create a new {@link FormatProcessor} using the specified locale.
+     *
+     * @param locale {@link Locale} used to format
+     *
+     * @return a new instance of {@link FormatProcessor}
+     *
+     * @throws java.lang.NullPointerException if locale is null
+     */
+    public static FormatProcessor create(Locale locale) {
+        Objects.requireNonNull(locale);
+        return new FormatProcessor(locale);
+    }
+
+    /**
+     * Constructs a {@link String} based on the fragments, format
+     * specifications found in the fragments and values in the
+     * supplied {@link StringTemplate} object. This method constructs a
+     * format string from the fragments, gathers up the values and
+     * evaluates the expression
+     * {@code new Formatter(locale).format(format, values).toString()}.
```

Comment:
Should this be described using the "as if"... phrasing to avoid a literal requirement in the spec that is inflexible?

Reply:
Changing